File: node/src/entities/number.ts
import { Entity, PrimaryKey, Property } from "@mikro-orm/core";

@Entity({
  tableName: 'numbers'
})
export class NumberEntity {
  @PrimaryKey()
  id!: number
  @Property()
  value!: number
  @Property({
    fieldName: 'is_prime'
  })
  isPrime!: boolean
}

File: rust/src/entities/number.rs
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel, Deserialize, Serialize)]
#[sea_orm(table_name = "numbers")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    pub value: i32,
    pub is_prime: bool,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}
